src/Requests.jl

```diff
@@ -153,6 +156,11 @@
         r.finished = true
         close(p.sock)
 
+        #If MIME type allows, convert r.data to a Julia string
+        mimetype = r.headers["Content-Type"]
+        if (contains(mimetype, "charset=utf-8") || contains(mimetype, "charset=us-ascii"))
```

Comment:
While I agree with all that, we should acknowledge a loss in convenience since probably 90% of the time in practice you do want a string. Maybe we should have a convenience method similar to `takebuf_string` but for `Response` objects.